src/proc.ts

```diff
@@ -146,6 +158,11 @@ export function execute(command: string, args?: string[], outputConsumer?: Outpu
             log.debug(localize('execution.environment', '  with environment: {0}', JSON.stringify(final_env)));
         }
     }
+
+    if (options.shell === undefined && process.platform === "win32") {
```

Comment:
should `undefined` be treated differently than `false`?

Reply:
Yes. If it's false, then we are explicitly saying that we don't want the shell, if it's `undefined`, in the Windows case, we now need to check if it's a ps1 or bat script and ensure we handle it accordingly.

Reply:
However, I am pushing a change to change the order of the if statement, we want to check if it's windows first

Reply:
Ah gotcha